framework/src/base/FEProblem.C

```diff
@@ -704,16 +709,47 @@ FEProblem::getMaxScalarOrder() const
 }
 
 void
+FEProblem::checkNonlocalCoupling()
```

Comment:
Sounds good.

Don't do it by doing subdomains * variables * kernels though... just loop through all of the Kernels and check them.